### PR TITLE
refactor: move types in `resolve` module to seperate files

### PR DESCRIPTION
**Pull request by tdejager**
It was getting hard to find the functions that you needed while browsing the module. So moved some types to their own files. No functionality has changed.

### CODE REVIEW

File: crates/rip_bin/src/main.rs [hunk 1]
```diff
@@ -97,18 +96,19 @@ struct SDistResolution {
     only_sdists: bool,
 }
 
-impl From<SDistResolution> for resolve::SDistResolution {
```
Comment (nichmor) [inline crates/rip_bin/src/main.rs:100]: wdyt about renaming the SDistResolution from src/main.rs? into something like: DistResolution? or maybe ResolutionArgs? so we will have a distinction between cli args and actually library option and will not hide the name

**Review by nichmor — APPROVED** [review]
Looks super clean!
only one small question

File: crates/rip_bin/src/main.rs (hunk 1 above)
Comment (tdejager) [reply crates/rip_bin/src/main.rs:100]: Yeah great idea! Let me change it!

Comment (tdejager) [reply crates/rip_bin/src/main.rs:100]: Done!

**Pull request merged**